File: 02-Sorting and Searching/Stick_Lengths.cpp
/*
 ____  _       _            ____              _ 
|  _ \(_)_   _(_)_ __   ___/ ___|  ___  _   _| |
| | | | \ \ / / | '_ \ / _ \___ \ / _ \| | | | |
| |_| | |\ V /| | | | |  __/___) | (_) | |_| | |
|____/|_| \_/ |_|_| |_|\___|____/ \___/ \__,_|_|                                      
                                                  
*/
#include <bits/stdc++.h>
#define ll long long
#define deb1(x) cout << #x << " " << x << "\n";
#define deb2(x, y) cout << #x << " " << x << " " << #y << " " << y << "\n";
#define deb3(x, y, z) cout << #x << " " << x << " " << #y << " " << y << " " << #z << " " << z << "\n";
#define ios()                \
    ios::sync_with_stdio(0); \
    cin.tie(0)
#define tc    \
    int t;    \
    cin >> t; \
    while (t--)
using namespace std;

void readFile()
{
#ifndef ONLINE_JUDGE
    freopen("input.txt", "r", stdin);
    freopen("output.txt", "w", stdout);
#endif
}

//----------------------------------------------------------------------------------
int main()
{
    //readFile();
    ios();
    int n;
    cin >> n;
    int t;
    vector<int> m;
    for (int i = 0; i < n; i++)
    {
        cin >> t;
        m.push_back(t);
    }
    sort(m.begin(), m.end());
    int mid;
    if(n%2==0){
         mid = m[n/2];
    }
    else {
         mid = (m[n/2] + m[n/2+1])/2;
    }
    ll ans = 0;
    for(int i=0; i<n; i++){
        ans += abs(mid-m[i]);
    }
    cout<<ans<<"\n";
    return 0;
}

File: 02-Sorting and Searching/Stick_Lengths.py
"""
 ____  _       _            ____              _ 
|  _ \(_)_   _(_)_ __   ___/ ___|  ___  _   _| |
| | | | \ \ / / | '_ \ / _ \___ \ / _ \| | | | |
| |_| | |\ V /| | | | |  __/___) | (_) | |_| | |
|____/|_| \_/ |_|_| |_|\___|____/ \___/ \__,_|_|                                      
                                                  
"""
n = int(input())
l = list(map(int,input().split()))
m = {}
for e in l:
    if e not in m.keys():
        m[e]=1
    else:
        m[e]+=1
m = sorted(m.items(),key = lambda x:x[1], reverse = True)
val = m[0][0]
ans = 0
for i in range(1,len(m)):
    ans += abs(m[i][0]-m[0][0])*m[i][1]
print(ans)  
